src/complex/DataStructure/Factory/NeighborListFactory.hpp

```diff
@@ -45,7 +82,76 @@ class COMPLEX_EXPORT NeighborListFactory : public IDataFactory
    * @return H5::ErrorType
    */
   H5::ErrorType readH5Dataset(H5::DataStructureReader& dataStructureReader, const H5::GroupReader& parentReader, const H5::DatasetReader& datasetReader,
-                              const std::optional<DataObject::IdType>& parentId = {}, bool preflight = false) override;
+                              const std::optional<DataObject::IdType>& parentId, bool preflight) override
```

Comment:
I think we should have the default arguments for parentId and preflight put back in. I'm sure it was just an oversight.